### PR TITLE
Github Actions added to pre-compile artifacts for flang

### PR DESCRIPTION
**Pull request by michalpasztamobica**
This PR is required in order to provide prebuilt artifacts for Flang project's CI as planned in https://github.com/flang-compiler/flang/pull/948.

The CI job needs to produce binary, precompiled artifacts to be used in Flang repository, os that the PR checks don't last for 3 hours, but 10 minutes, instead. The existing jobs in this repository do not provide this kind of artifacts. The job will only run after release_100 gets new commits, so that the latest build is always available for Flang job to use.

@kiranchandramohan , @RichBarton-Arm , please provide your review and please set up other reivewers appropriately. Thanks!

### CODE REVIEW

**Review by gklimowicz — APPROVED** [review]
I am not in a position to test this, so I will defer to the other reviewers. LGTM.

**Review by bryanpkc — APPROVED** [review]
LGTM. Thanks @michalpasztamobica.

**Comment by michalpasztamobica** [comment]
@shivaramaarao, are there any review remarks from your side?

**Review by shivaramaarao — APPROVED** [review]
Looks good. However I am not able to test this patch.

**Comment by michalpasztamobica** [comment]
Github actions are now enabled for this repository, I'll try to push this branch and see if that triggers them to test this patch for us :wink:

**Comment by michalpasztamobica** [comment]
Nope, a push didn't trigger the build, not sure why. @kiranchandramohan are you able to trigger the github actions for this build somehow?

**Comment by michalpasztamobica** [comment]
@shivaramaarao , I hope this script will produce [something of this kind](https://github.com/michalpasztamobica/classic-flang-llvm-project/actions/runs/354267060) once it's merged.

**Comment by kiranchandramohan** [comment]
@michalpasztamobica Not clear to me why it is not running. The CI was always running, the patch I added only limited it to the passing runs. In the case of this patch, it did not previously also run github actions.

**Comment by kiranchandramohan** [comment]
@michalpasztamobica The existing workflows have the following yaml at the beginning. Could you check whether it has something to the way a branch is named?

```
on:
  push:
    branches:
      - 'release/**'
...
```

**Comment by michalpasztamobica** [comment]
@kiranchandramohan , right - that's the problem. The job is scheduled to run on `push` event meaning - "commit to master" and not on pull requests. That's why it isn't running until it gets merged. This is intended - we just want to generate artifacts with this job, it doesn't really test anything.

Why the other jobs are not running is also because of how the yaml scripts are configured. They will only trigger if PR touches the paths:
```
  pull_request:
    paths:
      - 'clang/**'
      - 'llvm/**'
```

**Comment by michalpasztamobica** [comment]
@kiranchandramohan , sorry for the confusion, we won't see this job run until we merge and we won't see any jobs running for this PR because it doesn't change clang or llvm.
I tested how it runs within my private fork, so I expect no trouble here.
All reviewers approved, please let me know if anything is looking suspicious to you.

**Review by kiranchandramohan — APPROVED** [review]
LGTM.

**Pull request merged**
